src/pages/RoomInvitePage.js

```diff
@@ -82,22 +82,57 @@ function RoomInvitePage(props) {
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
+    const cleanedSearchTerm = useMemo(() => searchTerm.trim().replace(/\s*,\s*/g, ','), [searchTerm]);
+    const searchEmails = useMemo(() => _.compact(cleanedSearchTerm.split(',')), [cleanedSearchTerm]);
+
     useEffect(() => {
-        const inviteOptions = OptionsListUtils.getMemberInviteOptions(props.personalDetails, props.betas, searchTerm, excludedUsers);
-
-        // Update selectedOptions with the latest personalDetails information
-        const detailsMap = {};
-        _.forEach(inviteOptions.personalDetails, (detail) => (detailsMap[detail.login] = OptionsListUtils.formatMemberForList(detail, false)));
-        const newSelectedOptions = [];
-        _.forEach(selectedOptions, (option) => {
-            newSelectedOptions.push(_.has(detailsMap, option.login) ? {...detailsMap[option.login], isSelected: true} : option);
+        const emails = searchEmails;
+
+        // This ensures suggestions are displayed when there is no search term
+        // or the final list entry is empty
+        if (emails.length === 0 || cleanedSearchTerm.endsWith(',')) {
+            emails.push('');
+        }
+
+        const newUsersToInviteDict = {};
+        const newSelectedOptionsDict = {};
+
+        _.each(emails, (email, i) => {
```

Comment:
do we need the variable `i`? and if yes then can we rename it to `index`?

Reply:
Yes, it is used in the function body. I will rename